### ISSUE TITLE
[Vertical Taskbar for Windows 11] [Windows 11 Start Menu Styler] Start Menu width should be moved from vertical taskbar to start menu styler

### ISSUE THREAD

**Issue by bergamin** [ramensoftware/windhawk-mods#1013]
Author: @m417z

It doesn't make much sense to have this setting in the Vertical Taskbar mod as it doesn't style the menu, only the taskbar. Start Menu Styler on the other hand does style the start menu

**Comment by SandTechStuff** [comment]
This was added to the Vertical Taskbar mod to prevent clipping issues when using the Start Menu Styler mod. The Vertical Taskbar mod changed the start menu width before, I think it had something to do with putting the start menu at the top of the screen, it's just customizable now.

https://github.com/ramensoftware/windhawk-mods/discussions/882#discussioncomment-10517272

**Comment by m417z** [comment]
I agree that it's not intuitive and seems out of place, but this is a better-than-nothing option. The reason it's in the Vertical Taskbar mod is the following:

By default, the start menu container occupies the whole screen, and the start menu is drawn inside it:

![image](https://github.com/user-attachments/assets/68ab445d-67f1-4d35-8756-34d18138dd9c)

The container is always the same, but the drawn content moves depending on whether the taskbar icons are placed in the middle or on the left.

Now, the Vertical Taskbar mod had to move the start menu to another location. One way was to do change the drawn location. That'd probably the more correct way, but it's also more difficult to implement and maintain. An easier solution is to move and resize the container, in which case the drawn content adjusts automatically as it's just drawn in that container:

![image](https://github.com/user-attachments/assets/10f998cd-7073-4efd-8651-0e374bb88bb5)

Just like it's not trivial to move the drawn location, it's also not trivial to deduce its actual size, so the mod hardcodes the default start menu size. But if the start menu is customized, the size (mainly width) may be different depending on the theme. This better-than-nothing option allows to change the size of the container to fit the customized start menu.

Again, the better solution would be to have it done automatically one way or another, but I don't think it's worth the time and effort, especially since it can be invested in so many more important tasks, even related to this mod.

**Comment by bergamin** [comment]
I see. That explains this glitch

![image](https://github.com/user-attachments/assets/c133d926-315a-4944-9fdd-fe39c649fee5)

The shadow and the account card get truncated. That's probably the container's height

I thought I saw this being reported somewhere, but I can't find the issue anymore

**Comment by m417z** [comment]
I don't think it was reported, I haven't seen this issue before. Please create a new issue for it.

**Comment by SandTechStuff** [comment]
I included a small blurb in https://github.com/ramensoftware/windhawk-mods/discussions/882#discussioncomment-10516195, must have flown under the radar.

**Comment by bergamin** [comment]
@m417z 

> I don't think it was reported, I haven't seen this issue before. Please create a new issue for it.

#1015

@SandTechStuff 

> I included a small blurb in [#882 (reply in thread)](https://github.com/ramensoftware/windhawk-mods/discussions/882#discussioncomment-10516195), must have flown under the radar.

yeah, that's the one I saw

**Comment by m417z** [comment]
> One way was to change the drawn location. That'd probably be the more correct way, but it's also more difficult to implement and maintain.

That's now done in version 1.3.6. The option was removed.
